runtimes/pythonrt/runner_manager.go

```diff
@@ -48,37 +49,56 @@ type Healther interface {
 	Health(ctx context.Context, in *pb.HealthRequest, opts ...grpc.CallOption) (*pb.HealthResponse, error)
 }
 
-func waitForServer(name string, h Healther, timeout time.Duration) error {
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	defer cancel()
-	start := time.Now()
-	var req pb.HealthRequest
-
-	for time.Since(start) <= timeout {
-		resp, err := h.Health(ctx, &req)
-		if err != nil || resp.Error != "" {
-			time.Sleep(10 * time.Millisecond)
-			continue
-		}
-
-		return nil
+func waitForServer(ctx context.Context, sl *zap.SugaredLogger, name string, h Healther, timeout time.Duration) error {
+	if timeout == 0 {
+		timeout = 10 * time.Second
 	}
 
-	return fmt.Errorf("%s not ready after %v", name, timeout)
+	tmo := time.After(timeout)
+
+	const retryInterval = 50 * time.Millisecond
+
+	startTime := time.Now()
+
+	for {
```

Comment:
I think you can refactor this in a more readable way as a loop with max retries like 5 or 10 or whatever configurable times instead of multiple timeouts